### PR TITLE
Refactor torch util functions

### PR DESCRIPTION
**Pull request by irisliucy**
I combined `garage.torch._utils` and `garage.torch.utils` into `garage.torch._functions` and exported the utility functions to `garage.torch`.

### CODE REVIEW

File: docs/user/experiments.rst [hunk 1]
```diff
@@ -215,7 +215,7 @@ In order to enable the GPU for PyTorch, add the following code snippets to the e
 .. code-block:: python
 
   import torch
-  import garage.torch.utils as tu
+  import garage.torch._functions as tu
```
Comment (ryanjulian) [inline docs/user/experiments.rst:218]: these should be access from `garage.torch`

File: examples/torch/mtsac_metaworld_ml1_pick_place.py [hunk 2]
```diff
@@ -20,10 +20,10 @@
 from garage.experiment import deterministic, LocalRunner
 from garage.replay_buffer import PathBuffer
 from garage.sampler import LocalSampler
+import garage.torch._functions as tu
 from garage.torch.algos import MTSAC
 from garage.torch.policies import TanhGaussianMLPPolicy
 from garage.torch.q_functions import ContinuousMLPQFunction
```
Comment (ryanjulian) [inline examples/torch/mtsac_metaworld_ml1_pick_place.py:26]: please import these utils from garage.torch

File: src/garage/torch/algos/ddpg.py [hunk 3]
```diff
@@ -9,7 +9,7 @@
 from garage import _Default, make_optimizer
 from garage.np.algos.off_policy_rl_algorithm import OffPolicyRLAlgorithm
 from garage.replay_buffer import PathBuffer
-import garage.torch.utils as tu
+from garage.torch import dict_np_to_torch, torch_to_np
```
Comment (ryanjulian) [inline src/garage/torch/algos/ddpg.py:12]: yes -- please use all imports like this

File: src/garage/torch/__init__.py [hunk 4]
```diff
@@ -1,6 +1,21 @@
 """PyTorch-backed modules and algorithms."""
-from garage.torch._utils import compute_advantages
-from garage.torch._utils import filter_valids
-from garage.torch._utils import pad_to_last
+from garage.torch._functions import _DEVICE
```
Comment (ryanjulian) [inline src/garage/torch/__init__.py:2]: don't import _ variables -- those are private

Comment (irisliucy) [reply src/garage/torch/__init__.py:2]: Gotcha, will remove all private variables import.

**Comment by irisliucy** [comment]
@Merifyio rebase

**Comment by ryanjulian** [comment]
remember, rather than babysitting your PR, you can add the label `ready-to-merge` and Mergify will do it for you. this also helps the CI run faster (by not scheduling jobs which don't need to be run)
